# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/CSSEditUtils.h
# Commit: 3efbfdf889b4
# Full Hash: 3efbfdf889b4bcda432a73871eb329523f78357b
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2022-08-04 09:46:07
# Regressor Bug: 1774704
# File Overlap Count: 1
# Description:
#   Bug 1774704 - part 7-2: Make `CSSEditUtils::RemoveCSSInlineStyleWithTransaction` stop touching `Selection` directly r=m_kato
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152975
# ==============================================================================

diff -r f9d7d41f5043 -r 3efbfdf889b4 editor/libeditor/CSSEditUtils.h
--- a/editor/libeditor/CSSEditUtils.h	Thu Aug 04 04:19:46 2022 +0000
+++ b/editor/libeditor/CSSEditUtils.h	Thu Aug 04 04:28:13 2022 +0000
@@ -134,10 +134,12 @@
    * @param aProperty       [IN] The CSS property atom to remove.
    * @param aPropertyValue  [IN] The value of the property we have to remove
    *                             if the property accepts more than one value.
+   * @return                A candidate point to put caret.
    */
-  [[nodiscard]] MOZ_CAN_RUN_SCRIPT nsresult RemoveCSSInlineStyleWithTransaction(
-      nsStyledElement& aStyledElement, nsAtom* aProperty,
-      const nsAString& aPropertyValue);
+  [[nodiscard]] MOZ_CAN_RUN_SCRIPT Result<EditorDOMPoint, nsresult>
+  RemoveCSSInlineStyleWithTransaction(nsStyledElement& aStyledElement,
+                                      nsAtom* aProperty,
+                                      const nsAString& aPropertyValue);
 
   /**
    * Answers true is the property can be removed by setting a "none" CSS value